v23.1.4 RN (#17295)
cockroachdb/docs: src/current/_includes/releases/v23.1/v23.1.4.md

## v23.1.4

Release Date: June 20, 2023

{% include releases/release-downloads-docker-image.md release=include.release %}

<h3 id="v23-1-4-security-updates">Security updates</h3>

> - Introduced the [`sql.auth.createrole_allows_grant_role_membership.enabled`](../v23.1/cluster-settings.html#setting-sql-auth-createrole-allows-grant-role-membership-enabled) cluster setting, which defaults to `false`. If it is set to `true`, then users who have the [`CREATEROLE` role option](../v23.1/create-user.html#role-options) can grant and revoke role membership in any non-admin role. A future major version of CockroachDB will remove this cluster setting and make this the default behavior. [#104445][#104445]

Reviewer: ```suggestion
- The new [`sql.auth.createrole_allows_grant_role_membership.enabled`](../v23.1/cluster-settings.html#setting-sql-auth-createrole-allows-grant-role-membership-enabled) cluster setting allows a user with the [`CREATEROLE` role option](../v23.1/create-user.html#role-options) to grant and revoke non-admin roles. This cluster setting defaults to `false`, but is expected to become the default behavior in the future. [#104445][#104445]
```

<h3 id="v23-1-4-{{-site.data.products.enterprise-}}-edition-changes">{{ site.data.products.enterprise }} edition changes</h3>

> - Fixed an initialization race condition in [changefeed schema feeds](../v23.1/changefeed-examples.html) that could result in null pointer exceptions. [#104934][#104934]

Reviewer: ```suggestion
- Fixed an initialization race condition in [changefeed schema feeds](../v23.1/changefeed-examples.html) that could cause a node to crash with a null pointer exception. [#104934][#104934]
```

<h3 id="v23-1-4-sql-language-changes">SQL language changes</h3>

> - Users with [`MODIFYSQLCLUSTERSETTING`](../v23.1/security-reference/authorization.html#supported-privileges) will no longer be able to view non-`sql.defaults` [cluster settings](../v23.1/cluster-settings.html). [#104542][#104542]

Reviewer: I don't fully understand this and it seems important. Does this mean that even though you have permission to modify the setting, you can't see settings that have already been changed from their default?

<h3 id="v23-1-4-operational-changes">Operational changes</h3>

- The [`debug.zip`](../v23.1/cockroach-debug-zip.html) archive now contains the files formerly located at `nodes/*/ranges/*.json` in one file per node, `nodes/*/ranges.json`. [#104248][#104248]
- The `http-defaults` and `http-servers` sections of the [log config](../v23.1/configure-logs.html) will now accept a `headers` field containing a map of key-value string pairs which will comprise custom HTTP headers appended to every request. Additionally, a `compression` value can now be set to `gzip` or `none` to select a compression method for the HTTP request body. By default, `gzip` is selected. Previous functionality did not compress by default. [#104814][#104814]

<h3 id="v23-1-4-command-line-changes">Command-line changes</h3>

> - The new new log config option [`buffering`](../v23.1/configure-logs.html#log-buffering-for-network-sinks) allows you to [format buffer output as JSON arrays](../v23.1/configure-logs.html#file-logging-format). This is useful for APIs that consume JSON arrays, such as the Datadog logs API. [#104790][#104790]

Reviewer: ```suggestion
- The new log config option [`buffering`](../v23.1/configure-logs.html#log-buffering-for-network-sinks) allows you to [format buffer output as JSON arrays](../v23.1/configure-logs.html#file-logging-format). This is useful for APIs that consume JSON arrays, such as the Datadog logs API. [#104790][#104790]
```
This was probably my fault

> - There is a new log config option for the [`buffering` field](../v23.1/configure-logs.html#log-buffering-for-network-sinks) to [allow buffer output to be formatted as JSON arrays](../v23.1/configure-logs.html#file-logging-format). This is useful for APIs that require JSON array format, such as the Datadog logs API. [#104790][#104790]

Reviewer: ```suggestion
- The new new log config option [`buffering`](../v23.1/configure-logs.html#log-buffering-for-network-sinks) allows you to [format buffer output as JSON arrays](../v23.1/configure-logs.html#file-logging-format). This is useful for APIs that consume JSON arrays, such as the Datadog logs API. [#104790][#104790]
```

<h3 id="v23-1-4-db-console-changes">DB Console changes</h3>

> - The tenant dropdown is now hidden for users logged in to fewer than 2 tenants. If the singular tenant is not the system tenant, the dropdown remains visible. [#104819][#104819]

Reviewer: ```suggestion
- The tenant selection field is now visible only when a user is logged in to more than one non-system tenant.. [#104819][#104819]
```

> - Added warning to DB Console overview page when all nodes are running on the new version, but the cluster upgrade is not finalized. [#104874][#104874]

Reviewer: ```suggestion
- The DB Console overview page now displays a warning when all nodes are running on the new version but the cluster upgrade has not been finalized. [#104874][#104874]
```

> - The histogram window merge calculation now interpolates quantile values more accurately, and [**Metrics** charts](../v23.1/ui-overview.html#metrics) in the DB Console are smoother and more accurate. [#104815][#104815]

Reviewer: "quantile" should probably be in the allowed vocab list.

<h3 id="v23-1-4-bug-fixes">Bug fixes</h3>

> - Fixed a bug where a transaction retry during the backfill of the [**Jobs** table](../v23.1/ui-jobs-page.html#jobs-table) could result in job rows being missed. [#104757][#104757]

Reviewer: ```suggestion
- Fixed a bug where a transaction retry could miss job rows during the backfill of the [**Jobs** table](../v23.1/ui-jobs-page.html#jobs-table). [#104757][#104757]
```

> - Fixed a bug where admin or root user privileges were required to use [`SHOW SYSTEM GRANTS`](../v23.1/show-system-grants.html). [#104732][#104732]

Reviewer: ```suggestion
- Fixed a bug where admin or root user privileges were erroneously required to use [`SHOW SYSTEM GRANTS`](../v23.1/show-system-grants.html). [#104732][#104732]
```
Otherwise this is a bit ambiguous what the bug is.

> - Fixed a bug where the column selector on the [**Jobs** page](../v23.1/ui-jobs-page.html) was not visible. [#104738][#104738]

Reviewer: ```suggestion
- Fixed a bug that prevented display of the column selector on the [**Jobs** page](../v23.1/ui-jobs-page.html). [#104738][#104738]
```

> - Fixed a bug where invalid splits could crash and prevent restarts of nodes that hold a replica for the right-hand side. [#104850][#104850]

Reviewer: ```suggestion
- Fixed a bug where an invalid split could crash and prevent restarts of nodes that hold a replica for the right-hand side. [#104850][#104850]
```

> - Fixed the `debug recover make-plan` command to ignore partial range metadata when it can't be fully read, and instead rely solely on replica info from storage to produce the recovery plan. [#104774][#104774]

Reviewer: "Fixed a bug where...." perhaps, but what is "it can't be fully read"? The plan? The range?

> - Updated the histogram window merge calculation to more accurately interpolate quantile values. This change will result in smoother, more accurate [**Metrics** charts](../v23.1/ui-overview.html#metrics) in the DB Console. [#104815][#104815]

Reviewer: This doesn't seem like a bug, maybe move it to a different section.
```suggestion
- The histogram window merge calculation now interpolates quantile values more accurately, and [**Metrics** charts](../v23.1/ui-overview.html#metrics) in the DB Console are smoother and more accurate. [#104815][#104815]
```

> - Volumes that are likely to be duplicated in reported disk counters, such as RAID logical volumes, will no longer be double-counted. [#104806][#104806]

Reviewer: ```suggestion
- Fixed a metric bug that could cause volumes such as RAID logical volumes to be counted twice.[#104806][#104806]
```

> - Fixed a bug in `upstream etcd-io/raft` which could cause an unlimited amount of log to be pulled into memory, and lead to out-of-memory situations. The log scan now has a limited memory footprint. [#104968][#104968]

Reviewer: ```suggestion
- Fixed a bug in `upstream etcd-io/raft` which could cause an unlimited amount of log to be loaded into memory,. This could cause a node to crash with an out-of-memory (OOM) exception.The log scan now has a limited memory footprint. [#104968][#104968]
```

Reviewer: From looking at the upstream issue, I think that the upstream release note text might have a typo that changes its meaning 180 degrees. The upstream issue description is "release-23.1: sql: Limit MODIFYSQLCLUSTERSETTING to only view sql.defaults", which reads to me like only users with that setting can view sql.defaults. WDYT?

Author: > In 22.2, when a user sets the sql.auth.modify_cluster_setting_applies_to_all.enabled cluster setting to change the behavior of MODIFYCLUSTERSETTING system privilege, it should allow a user to only view and edit sql.defaults.*. Currently, while a user can edit sql.defaults.* they can view all cluster settings - this is not the expected behavior.
> In 23.1+, when a user has MODIFYSQLCLUSTERSETTING system privilege, it should allow a user to only view and edit sql.defaults.*. Currently, it shows all cluster settings - this is not the expected behavior (like Rafi alluded to previously).

From this description, I think the the problem was that users with that privilege could view all settings and now they can only view defaults.
